tutorial/markdown/python/python-langchain-pdf-chat/python-langchain-pdf-chat.md

```diff
@@ -0,0 +1,584 @@
+                }
+              ]
+            },
+            "metadata": {
+              "dynamic": true,
+              "enabled": true
+            },
+            "text": {
+              "enabled": true,
+              "dynamic": false,
+              "fields": [
+                {
+                  "index": true,
+                  "name": "text",
+                  "store": true,
+                  "type": "text"
+                }
+              ]
+            }
+          }
+        }
+      }
+    },
+    "store": {
+      "indexType": "scorch",
+      "segmentVersion": 16
+    }
+  },
+  "sourceType": "gocbcore",
+  "sourceName": "pdf-docs",
+  "sourceParams": {},
+  "planParams": {
+    "maxPartitionsPerPIndex": 64,
+    "indexPartitions": 16,
+    "numReplicas": 0
+  }
+}
+```
+
+### Setup Environment Config
+
+Copy the `secrets.example.toml` file in `.streamlit` folder and rename it to secrets.toml and replace the placeholders with the actual values for your environment. All configuration for communication with the database is read from the environment variables.
+
+```bash
+OPENAI_API_KEY = "<open_ai_api_key>"
+DB_CONN_STR = "<connection_string_for_couchbase_cluster>"
+DB_USERNAME = "<username_for_couchbase_cluster>"
+DB_PASSWORD = "<password_for_couchbase_cluster>"
+DB_BUCKET = "<name_of_bucket_to_store_documents>"
+DB_SCOPE = "<name_of_scope_to_store_documents>"
+DB_COLLECTION = "<name_of_collection_to_store_documents>"
+INDEX_NAME = "<name_of_fts_index_with_vector_support>"
+LOGIN_PASSWORD = "<password to access the streamlit app>"
+```
+
+> [OpenAI](https://openai.com) API Key is required for usage in generating embedding and querying LLM
+
+> The [connection string](https://docs.couchbase.com/python-sdk/current/howtos/managing-connections.html#connection-strings) expects the `couchbases://` or `couchbase://` part.
+
+> For this tutorial, `DB_BUCKET = pdf-chat`, `DB_SCOPE = _default`, `DB_COLLECTION = _default` and `INDEX_NAME = pdf_search`
+
+> Login_Password of Streamlit app is a basic password to access the app. You can set the password here and while using the app, password will be required to access the app.
+
+### Running the Application
+
+After starting couchbase server, adding vector index and installing dependencies. Our Application is ready to run.
+
+In the projects root directory, run the following command
+
+```sh
+streamlit run chat_with_pdf.py
+```
+
+The application will run on your local machine at http://localhost:8501.
+
+### Using PDF Chat App
+
+The page starts with a text box to enter your login password. This is the same password defined earlier in the [Setup Environment Config](#setup-environment-config) section. After submitting the password, the main application page will open.
+
+On the left sidebar, you'll find an option to upload a PDF document you want to use with this PDF Chat App. Depending on the size of the PDF, the upload process may take some time.
```

Comment:
Is a new line required after the text to better format the image?

Reply:
Same as above

Reply:
Agreed, Added extra line